framework/Util/TUtf8Converter.php

```diff
@@ -17,20 +17,81 @@
  *
  * @author Fabio Bas <fbio.bas@gmail.com>
  * @since 4.0.2
+ * @see https://en.wikipedia.org/wiki/ISO/IEC_2022 General structure of character encodings.
+ * @see https://en.wikipedia.org/wiki/ISO/IEC_646 National standards for ASCII.
+ * @see https://www.sljfaq.org/afaq/encodings.html Japanese encodings and character sets.
+ *
+ * These are not yet in iconv (as of April, 2023):
+ * @todo missing ISO-5427. https://en.wikipedia.org/wiki/ISO_5427. (8 bit Cyrillic, 1979/1981)
+ * @todo missing ČSN (Czech technical standard) 369103. https://en.wikipedia.org/wiki/KOI_character_encodings (also Cyrillic)
  */
 class TUtf8Converter
 {
+	public const ESC_CHARENCODINGS_MAP = [
+			"\x1B\x25\x47" => 'UTF-8', // ESC-'%G'
+
+			"\x1B\x28\x40" => 'ASCII',	"\x1B\x29\x40" => 'ASCII',	"\x1B\x2A\x40" => 'ASCII',	"\x1B\x2B\x40" => 'ASCII',
```

Comment:
I'm not sure this is the best class to put this array of "escape used to represent encodings", since it's outside the scope of UTF8 itself.
Still, it's the most matching class actually existing in prado, so unless you want to create another helper class, it's ok if they stay here by now.
Out of curiosity: it looks like each encoding has multiple escape codes, but i've not been able to understand why.

Reply:
I was debating if this is the right place to put it.  I think you may be right about this being in its own TEscCharsetConverter.

As for why there are 4 versions of each...  It has to do with G0, G1, G2, and G3 versions of each.  I do not know why there are G0-G3 versions or what each version does.   As far as I have been able to tell, they map the same way, but my experience is limited despite looking these things up.

Reply:
Now i get it, thank you